unzip/bsdunzip.c

```diff
@@ -694,8 +694,10 @@ extract_file(struct archive *a, struct archive_entry *e, char **path)
 		if (symlink(linkname, *path) != 0)
 			error("symlink('%s')", *path);
 		info(" extracting: %s -> %s\n", *path, linkname);
+#ifdef HAVE_LCHMOD
 		if (lchmod(*path, mode) != 0)
 			warning("Cannot set mode for '%s'", *path);
+#endif
```

Comment:
I guess we could move this up one line, and keep the warning when lchmod is not available.